Create payout modal

sauravmaiti22 opened the pull request:
1. Added modal to create payout.
2. Added validations as per design
3. Added atom for InputTextField to have a numeric variation to  accomdate  account number and sort code which are of a certain max length and are numeric. 
4. Separated out useCreatePayout from useCreateRefund hook as the invalidations are different.
5. Just has one button now to Create the payout, approve is WIP.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

agranillonf commented on packages/components/src/components/ui/molecules/Select/SelectBeneficiary/SelectBeneficiary.tsx:
```diff
@@ -0,0 +1,86 @@
+
+import { LocalBeneficiary } from '../../../../../types/LocalTypes'
+import { cn } from '../../../../../utils'
+// import { cn } from '../../../../../utils'
```
We can get rid of this comment.

sauravmaiti22 replied:
Done.

agranillonf commented:
There's some validation errors when clicking on "Create" without filling the required fields. It lets you go through and, if the API returns an error, the button remains in the "loading" state, unable to do anything unless the user refreshes the page.
![create_payout](https://github.com/nofrixion/nofrixion.business/assets/114026805/ee1b81fa-c377-45d4-a48e-bbb82c4796fb)

agranillonf commented:
The API requires an account name to be sent. If you don't input anything on it, it'll let you through but the API will return an error stating that. And same as above, the "Create" button will become stuck on the "loading" state.
![create_payout_2](https://github.com/nofrixion/nofrixion.business/assets/114026805/8a381e8e-b475-45a9-b13c-97ce1d57b250)

sauravmaiti22 commented:
Oops missed it. Thanks @agranillonf.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

agranillonf commented:
Required field validation has been fixed, but format doesn't seem like it.
![create_payout_3](https://github.com/nofrixion/nofrixion.business/assets/114026805/a62cdff6-0c89-4b89-904f-711f7d843da8)

sauravmaiti22 commented:
> Required field validation has been fixed, but format doesn't seem like it. ![create_payout_3](https://user-images.githubusercontent.com/114026805/266718748-a62cdff6-0c89-4b89-904f-711f7d843da8.gif) [ ![create_payout_3](https://user-images.githubusercontent.com/114026805/266718748-a62cdff6-0c89-4b89-904f-711f7d843da8.gif) ](https://user-images.githubusercontent.com/114026805/266718748-a62cdff6-0c89-4b89-904f-711f7d843da8.gif) [ ](https://user-images.githubusercontent.com/114026805/266718748-a62cdff6-0c89-4b89-904f-711f7d843da8.gif)

Yes I noticed that. Should be fixed now.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

donalnofrixion commented on packages/components/src/components/ui/InputAmountField/InputAmountField.tsx:
```diff
@@ -3,6 +3,7 @@ import * as DropdownMenu from '@radix-ui/react-dropdown-menu'
 import { cva } from 'class-variance-authority'
 import { motion } from 'framer-motion'
 import { InputHTMLAttributes, useEffect, useRef, useState } from 'react'
+// eslint-disable-next-line import/no-unresolved
```
Can you remove this?

donalnofrixion commented:
🔥 Looks great Saurav!

One small comment and not super urgent. Can you sort the accounts alphabetically please?
<img width="448" alt="Screenshot 2023-09-09 at 05 13 25" src="https://github.com/nofrixion/nofrixion.business/assets/89344328/a1a1f81e-ad01-43f3-b0e4-1aee4a27ff67">

pabloi-nf reviewed (changes requested):
The height of all fields have to be the same. Here the dropdown is higher than the text input field. The full height has to be 48px.
Also, the account name font-weight needs to be 400.
<img width="211" alt="image" src="https://github.com/nofrixion/nofrixion.business/assets/127946572/48df5cf9-a4c9-4810-89d2-eefad5642cf6">

The dropdown icon is not the correct one.
<img width="77" alt="image" src="https://github.com/nofrixion/nofrixion.business/assets/127946572/89967c3c-fc7f-4e9f-993c-c7e8cade6e81">
<img width="60" alt="image" src="https://github.com/nofrixion/nofrixion.business/assets/127946572/0a9a8282-982d-432e-8319-675399168e62">

The font-weight for all the information inside the beneficiary item has to be 400.
The font-size for the account number and currency has to be 0.75rem.
<img width="453" alt="image" src="https://github.com/nofrixion/nofrixion.business/assets/127946572/4374b7ed-3147-4060-b5e0-8fb7bf189842">

The "add manually" option needs to have a hover state without underline.

The non editable field background color has to be #F6F8F9.

We don't mark the required fields here, only the optional ones.

As far as I understand, the character limit for the reference is 17. Is that correct?

A put random data and I have an error that is not informative. What can be the situation here?
<img width="594" alt="image" src="https://github.com/nofrixion/nofrixion.business/assets/127946572/20308b8d-1d6a-475f-9de3-798067e062ee">

Let's remove the back arrow on top. It's better to keep that one for secondary modals, like the refund.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

sauravmaiti22 commented:
> The height of all fields have to be the same. Here the dropdown is higher than the text input field. The full height has to be 48px. Also, the account name font-weight needs to be 400. <img alt="image" width="211" src="https://user-images.githubusercontent.com/127946572/266784104-48df5cf9-a4c9-4810-89d2-eefad5642cf6.png">
> 
> The dropdown icon is not the correct one. <img alt="image" width="77" src="https://user-images.githubusercontent.com/127946572/266784314-89967c3c-fc7f-4e9f-993c-c7e8cade6e81.png"> <img alt="image" width="60" src="https://user-images.githubusercontent.com/127946572/266784332-0a9a8282-982d-432e-8319-675399168e62.png">
> 
> The font-weight for all the information inside the beneficiary item has to be 400. The font-size for the account number and currency has to be 0.75rem. <img alt="image" width="453" src="https://user-images.githubusercontent.com/127946572/266784354-4374b7ed-3147-4060-b5e0-8fb7bf189842.png">
> 
> The "add manually" option needs to have a hover state without underline.
> 
> The non editable field background color has to be #F6F8F9.
> 
> We don't mark the required fields here, only the optional ones.
> 
> As far as I understand, the character limit for the reference is 17. Is that correct?
> 
> A put random data and I have an error that is not informative. What can be the situation here? <img alt="image" width="594" src="https://user-images.githubusercontent.com/127946572/266784781-20308b8d-1d6a-475f-9de3-798067e062ee.png">
> 
> Let's remove the back arrow on top. It's better to keep that one for secondary modals, like the refund.

Thanks @pabloi-nf. I have fixed all the comments except for "We don't mark the required fields here, only the optional ones.". I couldn't understand this part. About the Create payout failing issue. It was because of invalid IBAN. I added more validation to IBAN now. It should show a message if it's not a valid IBAN.

Publicker commented:
> Thanks @pabloi-nf. I have fixed all the comments except for "We don't mark the required fields here, only the optional ones.". I couldn't understand this part. About the Create payout failing issue. It was because of invalid IBAN. I added more validation to IBAN now. It should show a message if it's not a valid IBAN.

Hey @sauravmaiti22, I believe @pabloi-nf  was getting at the idea that, unlike the 'Create Payment Request' form where only "Product or service" is marked as required, in this one we highlight the optional fields.

Check out the comparison between the actual implementation and the design:

Here's a comapartion between the implementation and the design:
| Implementation | Design |
|--------|--------|
| ![Implementation showing required fields](https://github.com/nofrixion/nofrixion.business/assets/52673485/b5e32ce4-3966-4d92-926e-32b58dc2a379) | ![Design - Showing optional fields](https://github.com/nofrixion/nofrixion.business/assets/52673485/fd5a01f3-8930-458c-a65d-394b9b9ffe67) | 

Hope that clears things up!

sauravmaiti22 commented:
> > Thanks @pabloi-nf. I have fixed all the comments except for "We don't mark the required fields here, only the optional ones.". I couldn't understand this part. About the Create payout failing issue. It was because of invalid IBAN. I added more validation to IBAN now. It should show a message if it's not a valid IBAN.
> 
> Hey @sauravmaiti22, I believe @pabloi-nf was getting at the idea that, unlike the 'Create Payment Request' form where only "Product or service" is marked as required, in this one we highlight the optional fields.
> 
> Check out the comparison between the actual implementation and the design:
> 
> Implementation	Design
> ![Implementation showing required fields](https://user-images.githubusercontent.com/52673485/267008608-b5e32ce4-3966-4d92-926e-32b58dc2a379.png)	![Design - Showing optional fields](https://user-images.githubusercontent.com/52673485/267008546-fd5a01f3-8930-458c-a65d-394b9b9ffe67.png)
> Hope that clears things up!

@Publicker Thanks Pablo. Yes I discussed it with @pabloi-nf and that was the case. He mentioned that in future we are going to change the implementation in create payment request as well.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

Publicker commented:
> @Publicker Thanks Pablo. Yes I discussed it with @pabloi-nf and that was the case. He mentioned that in future we are going to change the implementation in create payment request as well.

Brilliant! That makes sense!

Thanks @sauravmaiti.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

pabloi-nf reviewed (changes requested):
I'm seeing a weird animation when 2 warning messages are displayed and one of them dissapears. It seems one replaces the other somehow.

https://github.com/nofrixion/nofrixion.business/assets/127946572/50169edc-7b4f-49e7-81bf-5a632f5f3efb

I can see the "required" text appears before pressing the create button. This message must appear only after submiting, in connection with the "Please fill all required fields" message.
<img width="180" alt="image" src="https://github.com/nofrixion/nofrixion.business/assets/127946572/60d51a5a-2ba7-4223-9328-c81a7239838d">

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/57.
